integration-tests/src/test/java/tachyon/master/next/MasterClientIntegrationTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.next;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.thrift.TException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import tachyon.Constants;
+import tachyon.client.FileSystemMasterClient;
+import tachyon.conf.TachyonConf;
+
+/**
+ * Though its name indicates that it provides the tests for {@link tachyon.master.MasterClient},
+ * this class is more like unit-testing the internal implementation of tachyon Master via a
+ * {@link tachyon.master.MasterClient}, and thus it depends on many components in tachyon.master.
+ * As a result, we place MasterClient in tachyon-common and this test in tachyon-integration-tests.
+ *
+ * <p>
+ * TODO: Rename this class.
+ *
+ */
+public class MasterClientIntegrationTest {
```

Comment:
This should be FileSystemMasterClientIntegrationTest (and there would probably be another BlockMasterClientIntegrationTest).